src/mlpack/core/data/load_impl.hpp

```diff
@@ -410,6 +410,110 @@ bool LoadCategorical(const std::string& filename,
   return true;
 }
 
+template<typename MatType>
+bool Load(const std::vector<std::string>& filenames,
+          MatType& matrix,
+          const TextOptions& opts)
+{
+  TextOptions copyOpts(opts);
+  return Load(filenames, matrix, copyOpts);
+}
+
+template<typename MatType>
+bool Load(const std::vector<std::string>& filenames,
+          MatType& matrix,
+          TextOptions& opts)
+{
+  bool success = false;
+  MatType tmp;
+  arma::field<std::string> firstHeaders;
+  if (filenames.empty())
+  {
+    if (opts.Fatal())
+    {
+      Log::Fatal << "Load(): Please specify the filenames to be loaded!"
+          << std::endl;
+    }
+    else
+    {
+      Log::Warn << "Load(): Please specify the filenames to be loaded!"
+          << std::endl;
+      return false;
+    }
+  }
+  for (size_t i = 0; i < filenames.size(); ++i)
+  {
+    success = Load(filenames.at(i), matrix, opts);
+    if (opts.HasHeaders())
+    {
+      if (i == 0)
+        firstHeaders = opts.Headers();
+      else
+      {
+        arma::field<std::string>& headers = opts.Headers();
+        for (size_t j = 0; j < headers.size(); ++j)
+        {
+          if (firstHeaders.at(0) != headers.at(0))
```

Comment:
```suggestion
          if (firstHeaders.at(j) != headers.at(j))
```
If we want to check all the headers, we would need to do this.